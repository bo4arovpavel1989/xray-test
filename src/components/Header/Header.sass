.headerArea
  border-bottom: 1px solid grey

.header
  display: inline-block
  text-decoration: none

.header h1
  font-style: italic

.loginArea
  display: inline-block

.loginForm, .loginArea
  float: right
  position: relative

.logoff
  padding-left: 30px
  cursor: pointer
  color: blue

.authFailMessage
  position: absolute
  border: 1px solid red
  background-color: white
  top: 32px
  font-size: 10px
