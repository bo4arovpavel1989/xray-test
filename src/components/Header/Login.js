import React from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import './Header.sass'

const Login = props => {
  const { login, authFail, isLogging } = props

  return (
      <form className='loginForm' onSubmit = { login }>
        {
          authFail ?
            <div className='authFailMessage'>Неверное имя пользователя или пароль!</div>
          :
            ''
        }
        <span>
          <input type='login' name='login' placeholder='login'/>
          <input type='password' name='password' placeholder='password'/>
        </span>
        <span>
          <input disabled = { isLogging } type='submit' value='Войти'/>
        </span>
      </form>
  )
}

Login.propTypes = {
  login: PropTypes.func.isRequired
}

export default withRouter(Login)
