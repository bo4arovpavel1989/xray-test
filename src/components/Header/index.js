import React from 'react';
import {Link, withRouter} from 'react-router-dom';
import PropTypes from 'prop-types';
import Login from './Login';
import './Header.sass';

const Header = props=>{
  const {isAdmin, logoff, login} = props;

  return (
    <div className='headerArea'>
      <Link className='header' to='/'>
        <h1>XRay-test</h1>
      </Link>
      {
        isAdmin ?
          <div className='loginArea'>
            <a className='logoff' onClick={logoff}>
              Выйти
            </a>
          </div>
          :
          <Login login={login}/>
        }
    </div>
  )
};

Header.propTypes = {
  isAdmin: PropTypes.bool.isRequired,
  logoff: PropTypes.func.isRequired,
  login: PropTypes.func.isRequired
};

export default withRouter(Header);
