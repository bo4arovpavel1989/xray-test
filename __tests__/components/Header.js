import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import Header from '../../src/components/Header';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Header component', () => {
  const props = {
    isAdmin: false,
    authFail: false,
    login () {},
    logoff () {}
  };

  describe('Header snapshot', () => {
    const header = mount(
      <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
        <Header { ...props }/>
      </MemoryRouter>
    );

    it('should fit snapshot', () => expect(shallowToJson(header)).toMatchSnapshot())
    it('should render h1', () => expect(header.find('h1').text()).toEqual('XRay-test'))
    it('should render Login component', () => expect(header.find('Login')).toHaveLength(1))
  })

  describe('Header logged in', () => {
    const mockLogoff = jest.fn();

    props.isAdmin = true;
    props.logoff = mockLogoff;

    const header = mount(
      <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
        <Header { ...props }/>
      </MemoryRouter>
    );

    it('should fit snapshot', () => expect(shallowToJson(header)).toMatchSnapshot())
    it('should call logoff fn when clicking on logoff link', () => {
      header.find('.logoff').simulate('click');
      expect(mockLogoff).toHaveBeenCalledTimes(1);
    });
  })
})
