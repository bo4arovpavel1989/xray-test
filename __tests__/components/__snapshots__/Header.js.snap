// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header logged in should fit snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Header)
      authFail={false}
      isAdmin={true}
      login={[Function]}
      logoff={[MockFunction]}
    >
      <Route>
        <Header
          authFail={false}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isAdmin={true}
          location={
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            }
          }
          login={[Function]}
          logoff={[MockFunction]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="headerArea"
          >
            <Link
              className="header"
              replace={false}
              to="/"
            >
              <a
                className="header"
                href="/"
                onClick={[Function]}
              >
                <h1>
                  XRay-test
                </h1>
              </a>
            </Link>
            <div
              className="loginArea"
            >
              Вход выполнен
              <a
                className="logoff"
                onClick={[MockFunction]}
              >
                Выйти
              </a>
            </div>
          </div>
        </Header>
      </Route>
    </withRouter(Header)>
  </Router>
</MemoryRouter>
`;

exports[`Header component Header snapshot should fit snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Header)
      authFail={false}
      isAdmin={false}
      login={[Function]}
      logoff={[Function]}
    >
      <Route>
        <Header
          authFail={false}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isAdmin={false}
          location={
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            }
          }
          login={[Function]}
          logoff={[Function]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="headerArea"
          >
            <Link
              className="header"
              replace={false}
              to="/"
            >
              <a
                className="header"
                href="/"
                onClick={[Function]}
              >
                <h1>
                  XRay-test
                </h1>
              </a>
            </Link>
            <withRouter(Login)
              authFail={false}
              login={[Function]}
            >
              <Route>
                <Login
                  authFail={false}
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "key": "testKey",
                          "pathname": "/",
                          "search": "",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/",
                        "search": "",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    }
                  }
                  login={[Function]}
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <form
                    className="loginForm"
                    onSubmit={[Function]}
                  >
                    <span>
                      <input
                        name="login"
                        placeholder="login"
                        type="login"
                      />
                      <input
                        name="password"
                        placeholder="password"
                        type="password"
                      />
                    </span>
                    <span>
                      <input
                        type="submit"
                        value="Войти"
                      />
                    </span>
                  </form>
                </Login>
              </Route>
            </withRouter(Login)>
          </div>
        </Header>
      </Route>
    </withRouter(Header)>
  </Router>
</MemoryRouter>
`;
